numpy/__config__.py.in

```diff
@@ -1,134 +1,125 @@
-        for k,v in info_dict.items():
-            v = str(v)
-            if k == "sources" and len(v) > 200:
-                v = v[:60] + " ...\n... " + v[-60:]
-            print("    %s = %s" % (k,v))
-
-    features_found, features_not_found = [], []
-    for feature in __cpu_dispatch__:
-        if __cpu_features__[feature]:
-            features_found.append(feature)
-        else:
-            features_not_found.append(feature)
-
-    print("Supported SIMD extensions in this NumPy install:")
-    print("    baseline = %s" % (','.join(__cpu_baseline__)))
-    print("    found = %s" % (','.join(features_found)))
-    print("    not found = %s" % (','.join(features_not_found)))
+    if type(d) is dict:
+        return dict((k, _cleanup(v)) for k, v in d.items() if v and _cleanup(v))
+    else:
+        return d
+
+
+CONFIG = _cleanup(
+    {
+        "Compilers": {
+            "c": {
+                "name": "@C_COMP@",
+                "linker": "@C_COMP_LINKER_ID@",
+                "version": "@C_COMP_VERSION@",
+                "commands": "@C_COMP_CMD_ARRAY@",
+            },
+            "cython": {
+                "name": "@CYTHON_COMP@",
+                "linker": "@CYTHON_COMP_LINKER_ID@",
+                "version": "@CYTHON_COMP_VERSION@",
+                "commands": "@CYTHON_COMP_CMD_ARRAY@",
+            },
+            "c++": {
+                "name": "@CPP_COMP@",
+                "linker": "@CPP_COMP_LINKER_ID@",
+                "version": "@CPP_COMP_VERSION@",
+                "commands": "@CPP_COMP_CMD_ARRAY@",
+            },
+        },
+        "Machine Information": {
+            "host": {
+                "cpu": "@HOST_CPU@",
+                "family": "@HOST_CPU_FAMILY@",
+                "endian": "@HOST_CPU_ENDIAN@",
+                "system": "@HOST_CPU_SYSTEM@",
+            },
+            "build": {
+                "cpu": "@BUILD_CPU@",
+                "family": "@BUILD_CPU_FAMILY@",
+                "endian": "@BUILD_CPU_ENDIAN@",
+                "system": "@BUILD_CPU_SYSTEM@",
+            },
+            "target": {
+                "cpu": "@TARGET_CPU@",
+                "family": "@TARGET_CPU_FAMILY@",
+                "endian": "@TARGET_CPU_ENDIAN@",
+                "system": "@TARGET_CPU_SYSTEM@",
+            },
+        },
+        "Build Dependencies": {
+            "blas": {
+                "name": "@BLAS_NAME@",
+                "found": "@BLAS_FOUND@",
+                "version": "@BLAS_VERSION@",
+                "type name": "@BLAS_TYPE_NAME@",
+                "includedir": "@BLAS_INCLUDEDIR@",
+                "libdir": "@BLAS_LIBDIR@",
+                "openblas config": "@BLAS_OPENBLAS_CONFIG@",
+                "pc file directory": "@BLAS_PCFILEDIR@",
+            },
+            "lapack": {
+                "name": "@LAPACK_NAME@",
+                "found": "@LAPACK_FOUND@",
+                "version": "@LAPACK_VERSION@",
+                "type_name": "@LAPACK_TYPE_NAME@",
```

Comment:
Same here, "detection method"

Reply:
Refined in 72a73e6dc5a64230f69cd61707f6ddd18eaf935c